### ISSUE TITLE
Picker.ItemsSource does not update the picker after being manually set.

### ISSUE THREAD

**Issue by DuckScapePhilip** [dotnet/maui#9739]
### Description

I have a .NET MAUI project I am compiling for windows.

Before, the pickers were working when I was developing the project in Visual Studio 2022 - Release Preview.

Now, that I am developing in Visual Studio 2022, the pickers show empty. 

I have reproduced a similar issue in a scratch project. In this project, when a button is clicked, it will set the **picker.ItemSource** to a list of items. After this, in the application, you can see that the picker is still empty. If you click the button a second time, you will see the picker now has items.

### Steps to Reproduce

1. Get the project I uploaded to GitHub at https://github.com/DuckScapePhilip/Bugs/tree/main/ScratchPad.MAUI
2. Compile and run the project.
3. Notice the picker is empty.
4. Click the button called 
5. Notice the picker is still empty even though on line 23 in MainPage.xaml.cs we are setting the picker.ItemsSource
6. Click the button again
7. Notice the picker now has items in it.

### Version with bug

6.0.400

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

net6.0-windows10.0.19041.0

### Did you find any workaround?

I cannot find a workaround.

**Comment by jsuarezruiz** [comment]
Need to verify, but could be fixed by https://github.com/dotnet/maui/pull/9584

**Comment by DuckScapePhilip** [comment]
What I have figured out is that if:

- If the ItemsSource is NULL and then is set to a collection, it does not register.
- If the ItemsSource is set to an IList, and I set it to the same IList (same instance), it does not register.
- If the ItemsSource is set to an IList, and I set it to a different IList, it registers.

So my workaround was:
```
control.picker.ItemsSource = newValue as IList;

// TODO: Remove this as soon as MAUI has a bug fix. We have to do this to fix an issue with the MAUI picker.
control.picker.ItemsSource = control.picker.GetItemsAsArray();
control.picker.ItemsSource = newValue as IList;
```

**Comment by RealGregor** [comment]
> What I have figured out is that if:
> 
> * If the ItemsSource is NULL and then is set to a collection, it does not register.
> * If the ItemsSource is set to an IList, and I set it to the same IList (same instance), it does not register.
> * If the ItemsSource is set to an IList, and I set it to a different IList, it registers.
> 
> So my workaround was:
> 
> ```
> control.picker.ItemsSource = newValue as IList;
> 
> // TODO: Remove this as soon as MAUI has a bug fix. We have to do this to fix an issue with the MAUI picker.
> control.picker.ItemsSource = control.picker.GetItemsAsArray();
> control.picker.ItemsSource = newValue as IList;
> ```

I couldnt make it work until i removed the last line:
```
control.picker.ItemsSource = newValue as IList;
```

So now i have:
```
picker.ItemsSource = newValue;
picker.ItemsSource = picker.GetItemsAsArray();
```

**Comment by PureWeen** [comment]
@DuckScapePhilip I tested your sample with the latest .NET 7 bits and this issue looks like it'll be fixed. 

Just to confirm here's a screenshot I took running your sample. I only click the button once and the ItemSource fills up. 

![image](https://user-images.githubusercontent.com/5375137/188969815-e3b6544e-8b0f-48fd-8189-fdaed8b5dddc.png)

Am I following your steps correctly?

**Comment by MH-ZShearer** [comment]
I'm experiencing the same issue. I have a similar repo [here](https://github.com/MH-ZShearer/ItemsSourceUpdating).

Has the team been able to reproduce the issue using .NET 6? It appears from the most recent response the issue is resolved with .NET 7, however.

**Comment by DuckScapePhilip** [comment]
I have not tested this with .NET 7, but @MH-ZShearer there is a work around as seen above in my comments. So even though it is an issue in .NET 6, it's not critical since there is a work around.

If y'all say it is working in .NET 7, then I will take your word on it for now. Thanks for addressing this issue!

**Comment by DraganaBasaric** [comment]
I have the same problem.  
After I updated the new version of Visual Studio preview. 
My picker is now empty.  If I compile the App for ios, then everything works and the picker is not empty. 
For Windows machin  I used Code work around as seen above 
->ItemsSource is set to an IList, and I set it to a different IList, it registers.

Picker is on Windows machine not empty but now the event selectedIndexChange does not work for me.  
I need to be able to get in CodeBehind which item in the picker list is selected. I have no errors. If I debug, I see that the program does not execute the selectedIndexChange method at all.  

I would be very grateful if someone could help.
